src/Cpp/1-getting-started/1-HelloWindow/Input/Mouse.cpp

```diff
@@ -0,0 +1,104 @@
+#include "Mouse.hpp"
+#include "Input.hpp"
+
+#include <GLFW/glfw3.h>
+
+void Mouse::HandleMouseButtonPressed(
+    GLFWwindow* window,
+    const std::int32_t button,
+    const std::int32_t action,
+    const std::int32_t modifiers)
+{
+    _CRT_UNUSED(modifiers);
+    const auto input = static_cast<Input*>(glfwGetWindowUserPointer(window));
+    input->HandleKeyPressed(button, action);
+}
+
+void Mouse::HandleMousePosition(
+    GLFWwindow* window,
+    const double x,
+    const double y)
+{
+    const auto input = static_cast<Input*>(glfwGetWindowUserPointer(window));
+    input->HandleMouseMove(x, y);
+}
+
+Mouse::Mouse(): CursorPosition(), DeltaPosition(), _isCaptured(false)
+{
+}
+
+void Mouse::HandleButton(
+    const std::int32_t button,
+    const std::int32_t action)
+{
+    // ReSharper disable once CppDefaultCaseNotHandledInSwitchStatement
```

Comment:
Comment seems out of place?

Reply:
I believe clang has similar linting hints and shall be switched to once we figured clang out :)

For the time being resharper analysis hints remain